src/docs/process-binding.md

```diff
@@ -0,0 +1,87 @@
+# The Process of Binding
+
+In theory, the process of binding a Swift binary into C# should be as simple as:
+
+1. Generate and consume the abi.json file
+2. Extract and demangle the symbols from the binary
+3. Iterate over every type and function and gnerate C# and/or supporting Swift code
+4. Generate type datatbase entries
+5. Optionally compile generated code
+
+Unfortunately, it is not so simple. There are complications that need to be considered because of special cases in the Swift language as well as special cases or limitations in C#.
+
+For example
+- `open` Swift classes need to be handled differently from `public` swift classes.
+- Structs and enums that are `@frozen` are passed differently from structs that are not. 
+- Structs and enums that contain or may contain non-blitable members need to be defined differently.
+- Enums come in different forms that lend themselves to different representations in C#.
+- C# doesn't have methods or properties on enums so those need to be put into extension methods.
+- Any protocol implementation in C# will need proxy types in both C# and Swift
+- P/invokes and `[UnmanagedCallersOnly]` entrypoints (reverse p/invokes) need to be defined in a separate class from the bound type they support if that bound type is generic.
+- Mutating methods on value types need to be handled differently from non-mutating methods.
+- C# does not support methods that are homonyms.
+
+All of this makes the process of generating code challenging.
+
+The complexities fall into several broad categories:
```

Comment:
Do those complexities impose additional steps over the five you described, or they just make the steps more complicated?